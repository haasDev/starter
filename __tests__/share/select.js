import React from 'react';
import {mount} from 'enzyme';

import Select from '../../src/share/select';
import snapshot from '../snapshot';

describe('Select', () => {
  const disabled = false;
  const multiple = false;
  const onChange = jest.fn();
  const options = [
    {text: 'A', value: 'a'},
    {text: 'B', value: 'b'},
    {text: 'C', value: 'c'}
  ];
  const size = 5;
  const value = 'hello';
  const props = {disabled, multiple, onChange, options, size, value};

  it('should render', () => {
    snapshot(<Select {...props} />);
  });

  it('should call onChange when the select changes', () => {
    const wrapper = mount(<Select {...props} />);
    wrapper.find('select').simulate('change', {target: {value: 'world'}});
    expect(onChange).toHaveBeenCalled();
  });
});
