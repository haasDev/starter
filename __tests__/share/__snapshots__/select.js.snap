exports[`Select should render 1`] = `
<div
  className="select">
  <select
    className="form-control"
    disabled={false}
    multiple={false}
    onChange={[Function]}
    size={5}
    value="hello">
    <option
      value="a">
      A
    </option>
    <option
      value="b">
      B
    </option>
    <option
      value="c">
      C
    </option>
  </select>
</div>
`;
